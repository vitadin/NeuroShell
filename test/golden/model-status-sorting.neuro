%% Test model status sorting functionality
\set[_echo_command="true"]

%% Create models with different names and providers to test sorting
\model-new[catalog_id="O3"] zebra-model
\model-new[catalog_id="O4M"] alpha-model
\model-new[provider="openai", base_model="gpt-4"] middle-model

%% Test sorting by name (alphabetical)
\model-status[sort="name"]

%% Test sorting by provider (then by name)
\model-status[sort="provider"]

%% Test sorting by creation date (chronological)
\model-status[sort="created"]

%% Test invalid sort option (should show error)
\try \model-status[sort="invalid"]

%% Verify variables are still consistent
\get[#model_count]