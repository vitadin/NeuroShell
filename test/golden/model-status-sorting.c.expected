Setting _echo_command = true
%%> "\\model-new[catalog_id=\"O3R\"] zebra-model"
Created model 'zebra-model' (ID: 00000001, Provider: openai, Base: o3)
%%> "\\silent \\openai-client-new[client_type=OAR]"
%%> "\\silent \\llm-client-activate OAR"
%%> "\\silent \\model-activate[id=true] 00000001-0000-4000-8000-000000000001"
%%> "\\model-new[catalog_id=\"O4MC\"] alpha-model"
Created model 'alpha-model' (ID: 00000002, Provider: openai, Base: o4-mini)
%%> "\\silent \\openai-client-new[client_type=OAC]"
%%> "\\silent \\llm-client-activate OAC"
%%> "\\silent \\model-activate[id=true] 00000002-0000-4000-8000-000000000002"
%%> "\\model-new[catalog_id=\"G41C\"] middle-model"
Created model 'middle-model' (ID: 00000003, Provider: openai, Base: gpt-4.1-2025-04-14)
%%> "\\silent \\openai-client-new[client_type=OAC]"
%%> "\\silent \\llm-client-activate OAC"
%%> "\\silent \\model-activate[id=true] 00000003-0000-4000-8000-000000000003"
%%> "\\model-status[sort=\"name\"]"
Model Configurations (3 models):
  alpha-model (00000002)
    Provider: openai
    Base Model: o4-mini
    Catalog ID: O4MC
    Parameters: 1 (temperature=1)
    Created: 2025-01-01 00:00:02
    Description: OpenAI O4-mini Chat model

  middle-model (00000003)
    Provider: openai
    Base Model: gpt-4.1-2025-04-14
    Catalog ID: G41C
    Parameters: 1 (temperature=1)
    Created: 2025-01-01 00:00:03
    Description: OpenAI GPT-4.1 Chat model
    Active: true

  zebra-model (00000001)
    Provider: openai
    Base Model: o3
    Catalog ID: O3R
    Parameters: 2 (reasoning_effort=medium, reasoning_summary=auto)
    Created: 2025-01-01 00:00:01
    Description: OpenAI O3 Reasoning model

%%> "\\model-status[sort=\"provider\"]"
Model Configurations (3 models):
  alpha-model (00000002)
    Provider: openai
    Base Model: o4-mini
    Catalog ID: O4MC
    Parameters: 1 (temperature=1)
    Created: 2025-01-01 00:00:02
    Description: OpenAI O4-mini Chat model

  middle-model (00000003)
    Provider: openai
    Base Model: gpt-4.1-2025-04-14
    Catalog ID: G41C
    Parameters: 1 (temperature=1)
    Created: 2025-01-01 00:00:03
    Description: OpenAI GPT-4.1 Chat model
    Active: true

  zebra-model (00000001)
    Provider: openai
    Base Model: o3
    Catalog ID: O3R
    Parameters: 2 (reasoning_effort=medium, reasoning_summary=auto)
    Created: 2025-01-01 00:00:01
    Description: OpenAI O3 Reasoning model

%%> "\\model-status[sort=\"created\"]"
Model Configurations (3 models):
  zebra-model (00000001)
    Provider: openai
    Base Model: o3
    Catalog ID: O3R
    Parameters: 2 (reasoning_effort=medium, reasoning_summary=auto)
    Created: 2025-01-01 00:00:01
    Description: OpenAI O3 Reasoning model

  alpha-model (00000002)
    Provider: openai
    Base Model: o4-mini
    Catalog ID: O4MC
    Parameters: 1 (temperature=1)
    Created: 2025-01-01 00:00:02
    Description: OpenAI O4-mini Chat model

  middle-model (00000003)
    Provider: openai
    Base Model: gpt-4.1-2025-04-14
    Catalog ID: G41C
    Parameters: 1 (temperature=1)
    Created: 2025-01-01 00:00:03
    Description: OpenAI GPT-4.1 Chat model
    Active: true

%%> "\\try \\model-status[sort=\"invalid\"]"
%%> "\\model-status[sort=\"invalid\"]"
%%> "\\get[#model_count]"
#model_count = 3